### PR TITLE
Fix finding log in Load Dataset

### LINKED ISSUE #1649: NoneType error when loading phantom spheres dataset
### Summary

Trying to load the phantom spheres data as a dataset throws an error. Loading the flowers dataset still seems to work OK.

### Steps To Reproduce

1) Click File -> Load dataset
2) Attempt to load the phantom spheres test dataset

### Expected Behaviour

Load dialog is populated without error and the data can be loaded. If it's no longer supposed to be possible to load phantom spheres as a dataset due to the data structure then a helpful error message should be shown.

### Current Behaviour

An error message pops up with:

> The object passed as path for this field is not a string. Instead got <class 'NoneType'>

After clicking OK twice to dismiss the error, the OK button in the Load dialog is still disabled so it's not possible to load the dataset.

### Context

Current main

### Failure Logs

```
2022-12-16 14:37:20,983 [mantidimaging.gui.mvp_base.presenter:L24] ERROR: Presenter error: The object passed as path for this field is not a string. Instead got <class 'NoneType'>
Traceback (most recent call last):
  File "mantidimaging\mantidimaging\gui\windows\image_load_dialog\presenter.py", line 105, in do_update_sample
    self.set_sample_log(self.view.fields["Sample Log"], self.last_file_info.filenames)
  File "mantidimaging\mantidimaging\gui\windows\image_load_dialog\presenter.py", line 204, in set_sample_log
    sample_log.path = sample_log_filepath
  File "mantidimaging\mantidimaging\gui\windows\image_load_dialog\field.py", line 72, in path
    raise RuntimeError(f"The object passed as path for this field is not a string. Instead got {type(value)}")
RuntimeError: The object passed as path for this field is not a string. Instead got <class 'NoneType'>
Traceback (most recent call last):
  File "mantidimaging\mantidimaging\gui\windows\image_load_dialog\presenter.py", line 105, in do_update_sample
    self.set_sample_log(self.view.fields["Sample Log"], self.last_file_info.filenames)
  File "mantidimaging\mantidimaging\gui\windows\image_load_dialog\presenter.py", line 204, in set_sample_log
    sample_log.path = sample_log_filepath
  File "mantidimaging\mantidimaging\gui\windows\image_load_dialog\field.py", line 72, in path
    raise RuntimeError(f"The object passed as path for this field is not a string. Instead got {type(value)}")
RuntimeError: The object passed as path for this field is not a string. Instead got <class 'NoneType'>
2022-12-16 14:37:20,986 [mantidimaging.core.io.utility:L171] INFO: Could not find a log file for \phantom_spheres_float32\output\flat_0000.tiff
```

### PR DESCRIPTION
**Pull request by samtygier-stfc**
### Issue

Close #1649

### Description

Finish up some missing pathlib parts from #1636 . Also need checks in `do_update_sample()` because `find_log_for_image()` can return None.

Plus some extra type annotations that would have caught this issue.

### Testing & Acceptance Criteria 

Steps on #1649 should work without error

### Documentation

Not needed

### CODE REVIEW

File: mantidimaging/gui/windows/image_load_dialog/presenter.py [hunk 1]
```diff
@@ -110,8 +110,12 @@ def do_update_sample(self):
         self.view.fields["Sample Log"].use = False
 
         for pos in ["Before", "After"]:
-            self.view.fields[f"Flat {pos} Log"].path = find_log_for_image(self.view.fields[f"Flat {pos}"].path_text())
```
Comment (JackEAllen) [inline mantidimaging/gui/windows/image_load_dialog/presenter.py:113]: NITPICK: Given this is already a pretty long method and we are touching it anyway, what are you're thoughts on breaking this out into a separate method to be called called something like: `flat_path_logs` and break into a separate test in `presenter_test.py`?

**Review by JackEAllen — CHANGES_REQUESTED** [review]
Looks pretty good! Thank you for adding all the type checks Sam!

From manual testing, comparing between `main` and `1649-load-nonetype` follwing instructions left on #1689 loading the phantom spheres dataset no longer produces previously experienced errors relating to unexpected path types.

I have left a minor nitick I wanted to get your opinion on to seize an opportunity for a small refactor to shrink the size of `do_update_sample()` before approving. Once I have your opinion on this, happy to merge 👍

File: mantidimaging/gui/windows/image_load_dialog/presenter.py (hunk 1 above)
Comment (samtygier-stfc) [reply mantidimaging/gui/windows/image_load_dialog/presenter.py:113]: I'm a bit torn on this. I want to refactor, but I'll do it in future PRs

Comment (JackEAllen) [reply mantidimaging/gui/windows/image_load_dialog/presenter.py:113]: That's fine by me! It's why I wanted to get your opinion on this first. I'll approve this and merge in. We should create and issue to refactor and link to this PR for reference.

**Review by JackEAllen — APPROVED** [review]
LGTM! 👍

**Pull request merged**